### PR TITLE
new tests for project

### PR DESCRIPTION
**Pull request by MuyanjaKevin**
These are some of the changes i have made and some more tests i made

- testing for login
- testing for registeration with correct credentials
- testing registeration with incorrect credentials
- testing that login form appears after join events express button is clicked
- testing redirection when different links are clicked e.g about,contact,terms ,etc

### CODE REVIEW

File: config.py [hunk 1]
```diff
@@ -3,6 +3,19 @@
...
+CONTACT_US_PAGE='https://eventsexpress-test.azurewebsites.net/contactAdmin'
+ADMIN_EMAIL = "someemail@email.com"
 ADMIN_PASS = "somepassword"
+
+INVALID_REGISTER_EMAIL = 'k23@gmail.com'
+INVALID_REGISTER_PASSWORD = '123'
+
+USER_EMAIL = 'innatester601@gmail.com'
+USER_PASSWORD = 'innatester601'
+
```
Comment (mehalyna) [inline config.py:20]: don't push any credentials in github repo

File: config.py [hunk 2]
```diff
@@ -3,6 +3,19 @@
 """
 
 BASE_URL = "https://eventsexpress-test.azurewebsites.net/"
-ADMIN_EMAIL = "some@email.com"
+EVENTS_PAGE_URL = 'https://eventsexpress-test.azurewebsites.net/home/events'
+JOIN_EVENT_URL = 'https://eventsexpress-test.azurewebsites.net/home/events/8d58c4dc-0d43-44bd-4e71-08da439568bd'
```
Comment (mehalyna) [inline config.py:7]: we don't need the list of urls here

File: src/pages/LandingPage.py [hunk 3]
```diff
@@ -12,11 +12,41 @@ class LandingPage(BaseWrapper):
     CREATE_EVENT_BTN_CSS = "div.buttons > button"
     JOIN_EVENTSEXPRESS_BTN_CSS = "div.text-center > div.d-inline-block > button"
     LOG_OUT_BTN_CSS = "div.text-right > div"
+    EXPLORE_MORE_EVENTS_BTN_CSS = '#root > div > div > article.events-article > div.row > div.col-md-2 > a'
+    JOIN_EVENT_CSS = '#root > div > div > article.events-article > div.carousel-wrapper.text-center > div > div > div > div > div > div > div > div:nth-child(2) > a'
+    PRIVACY_BUTTON_CSS = '#root > div > footer > div.links-to-pages > a:nth-child(1)'
+    TERMS_BUTTON_CSS = '#root > div > footer > div.links-to-pages > a:nth-child(2)'
+    ABOUT_BUTTON_CSS = '#root > div > footer > div.links-to-pages > a:nth-child(3)'
+    CONTACT_US_BUTTON_CSS = '#root > div > footer > div.links-to-pages > a:nth-child(4)'
```
Comment (mehalyna) [inline src/pages/LandingPage.py:20]: Try to shorten selectors

File: src/pages/LandingPage.py [hunk 4]
```diff
@@ -12,11 +12,41 @@ class LandingPage(BaseWrapper):
     CREATE_EVENT_BTN_CSS = "div.buttons > button"
     JOIN_EVENTSEXPRESS_BTN_CSS = "div.text-center > div.d-inline-block > button"
     LOG_OUT_BTN_CSS = "div.text-right > div"
+    EXPLORE_MORE_EVENTS_BTN_CSS = '#root > div > div > article.events-article > div.row > div.col-md-2 > a'
```
Comment (mehalyna) [inline src/pages/LandingPage.py:15]: eg.
.events-article>div>div:last-of-type>a

File: src/pages/LandingPage.py [hunk 5]
```diff
@@ -12,11 +12,41 @@ class LandingPage(BaseWrapper):
...
+        return self.find_element_by_css(self.PRIVACY_BUTTON_CSS)
+
+    def get_terms_link(self):
+        return self.find_element_by_css(self.TERMS_BUTTON_CSS)
+
+    def get_about_link(self):
+        return  self.find_element_by_css(self.ABOUT_BUTTON_CSS)
+
+    def get_contact_us_link(self):
+        return self.find_element_by_css(self.CONTACT_US_BUTTON_CSS)
```
Comment (mehalyna) [inline src/pages/LandingPage.py:52]: suppose we have to define FooterPartPage and define all these methods here.
Consequently we'll use it throughout all pages

File: src/pages/LandingPageForAuthorizedUser.py [hunk 6]
```diff
@@ -25,3 +25,7 @@ def get_user_name(self):
         return self.find_element_by_css(self.USER_NAME_BTN_CSS).text
 
 
+
+
+
+
```
Comment (mehalyna) [inline src/pages/LandingPageForAuthorizedUser.py:31]: delete these rows, pls

File: tests/conftest.py [hunk 7]
```diff
@@ -25,6 +25,6 @@ def admin_setup(app):
         Login as an admin
     """
     app.go_to_site()
-    app.landing.sign_up_btn.click_btn_by_css()
-    app.modal.login(config.ADMIN_EMAIL, config.ADMIN_PASS)
+    app.landing.sign_in_up_btn.click_btn_by_css()
+    app.modal.login(config.USER_EMAIL, config.USER_PASSWORD)
```
Comment (mehalyna) [inline tests/conftest.py:29]: create one more method for the authorization of ordinary users. Don't change this one

File: tests/testScripts/test_landing.py [hunk 8]
```diff
@@ -2,29 +2,96 @@
...
     """
-    expected_result = "Your registration was successfully. Please confirm your email."
+    expected_result = "Your register was successfull. Please confirm your email."
     app.landing.go_to_site()
-    app.landing.sign_up_btn.click_btn_by_css()
+    app.landing.sign_in_up_btn.click_btn_by_css()
     app.modal.registration(config.ADMIN_EMAIL, config.ADMIN_PASS)
     assert expected_result == app.modal.get_success_register_text(), \
         "alert message is not the same as expected"
+
```
Comment (mehalyna) [inline tests/testScripts/test_landing.py:33]: don't change the functionality and purpose of this method. Create one more instead

File: tests/testScripts/test_landing.py [hunk 9]
```diff
@@ -2,29 +2,96 @@
...
+
+def test_landing_registeration_with_invalid_password(app):
+    """
+    verify the user needs to input a certain number of character for a password to be valid
+    """
+    expected_result = "Must be 6 characters or more"
+    app.landing.go_to_site()
+    app.landing.sign_in_up_btn.click_btn_by_css()
+    app.modal.registration(config.INVALID_REGISTER_EMAIL, config.INVALID_REGISTER_PASSWORD)
```
Comment (mehalyna) [inline tests/testScripts/test_landing.py:42]: in config you may use only constants with valid data. All other data are test data. We'll use them as from files or define them just here, in TC

File: tests/testScripts/test_landing.py [hunk 10]
```diff
@@ -2,29 +2,96 @@
...
+    assert expected_result
+    time.sleep(10)
+
+
+def test_about_link_redirects_user_to_about_page(app):
+    app.landing.go_to_site()
+    app.landing.about_link.click_btn_by_css()
+    expected_result = config.ABOUT_PAGE_URL
+    assert expected_result
+    time.sleep(10)
```
Comment (mehalyna) [inline tests/testScripts/test_landing.py:89]: we cannot use urls for assertions. We have to verify if we were redirected to the correct page. So try to handle any text or label on it

File: config.py [hunk 11]
```diff
@@ -3,6 +3,19 @@
...
+PRIVACY_PAGE_URL = 'https://eventsexpress-test.azurewebsites.net/privacy'
+TERMS_PAGE_URL = 'https://eventsexpress-test.azurewebsites.net/terms'
+ABOUT_PAGE_URL = 'https://eventsexpress-test.azurewebsites.net/about'
+CONTACT_US_PAGE='https://eventsexpress-test.azurewebsites.net/contactAdmin'
+ADMIN_EMAIL = "someemail@email.com"
 ADMIN_PASS = "somepassword"
+
+INVALID_REGISTER_EMAIL = 'k23@gmail.com'
+INVALID_REGISTER_PASSWORD = '123'
```
Comment (Thoughtseize1) [inline config.py:16]: remove it from config.py and push invalid data in TC.

**Review by servettonga — CHANGES_REQUESTED** [review]
Your base is old. Merge changes to your branch first.

File: src/pages/FooterPage.py
Comment (servettonga) [inline src/pages/FooterPage.py:1]: Work on Footer.py instead.

File: src/pages/PrivacyPage.py [hunk 12]
```diff
@@ -0,0 +1,17 @@
...
+    """
+    privacy page locators
+    """
+    PRIVACY_HEADING_CSS = '#main > div > h1'
+
+    def __init__(self, driver):
+        super().__init__(driver)
+
+    def get_privacy_page_heading(self):
+        return self.find_element_by_css(self.PRIVACY_HEADING_CSS).text
```
Comment (Thoughtseize1) [inline src/pages/PrivacyPage.py:15]: Here you also can create a variables with object and text

**Review by Thoughtseize1 — CHANGES_REQUESTED** [review]

File: src/pages/PrivacyPage.py [hunk 13]
```diff
@@ -0,0 +1,17 @@
+from src.common.BaseWrapper import BaseWrapper
+from src.webElements.ButtonElement import ButtonElement
+
+
+class Privacy_Page(BaseWrapper):
+    """
+    privacy page locators
+    """
```
Comment (Thoughtseize1) [inline src/pages/PrivacyPage.py:8]: Make docstrin in one line

File: src/pages/PrivacyPage.py [hunk 14]
```diff
@@ -0,0 +1,17 @@
...
+    PRIVACY_HEADING_CSS = '#main > div > h1'
+
+    def __init__(self, driver):
+        super().__init__(driver)
+
+    def get_Privacy_page_heading(self):
+        return self.find_element_by_css(self.PRIVACY_HEADING_CSS).text
```
Comment (Thoughtseize1) [inline src/pages/PrivacyPage.py:15]: You also can make var for it

File: tests/conftest.py [hunk 15]
```diff
@@ -27,34 +28,42 @@ def app():
...
+    app.modal.login(config.ADMIN_EMAIL, config.ADMIN_PASS)
+    return app
 
 
 @pytest.hookimpl(tryfirst=True, hookwrapper=True)
 def pytest_runtest_makereport(item):
-    """Hook the "item" object on a test failure"""
+    """
+        Hook the "item" object on a test failure
+    """
```
Comment (Thoughtseize1) [inline tests/conftest.py:45]: Shorter docstring would be better - https://www.datacamp.com/tutorial/docstrings-python

File: tests/testScripts/test_landing.py [hunk 16]
```diff
@@ -130,3 +130,11 @@ def test_sign_up_with_incorrect_data(app) -> None:
...
     error_msg = app.modal.find_element_by_xpath(app.modal.UNSUCCESS_PAGE_ALERT_TEXT_XPATH)
     assert error_msg.text == expected_result
+
+
+def test_user_is_redirected_to_privacy_page_when_privacy_link_clicked(app):
+    app.landing.go_to_site()
+    app.landing.scroll_down_page()
+    app.footer.privacy_link.click_btn_by_xpath()
+    expected_result = 'Privacy Policy'
+    assert expected_result == app.privacy.get_Privacy_page_heading()
```
Comment (Thoughtseize1) [inline tests/testScripts/test_landing.py:140]: Looks good but would be better if you write the last 2 string into one. 
`assert app.landing_authorized_user.drop_down_menu`

**Review by Thoughtseize1 — CHANGES_REQUESTED** [review]
Make changes )

**Pull request closed without merging**